mindsdb/integrations/utilities/rag/rerankers/reranker_async.py

```diff
@@ -0,0 +1,428 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+import math
+import os
+import random
+from typing import Any, Dict, List, Optional, Sequence, Tuple
+
+from openai import AsyncOpenAI, AsyncAzureOpenAI
+
+from mindsdb.integrations.utilities.rag.settings import DEFAULT_RERANKING_MODEL, DEFAULT_LLM_ENDPOINT
+
+log = logging.getLogger(__name__)
+
+
+class Document:
+    """A simple Document class to represent documents with content and metadata."""
+    def __init__(self, page_content: str, metadata: Optional[Dict[str, Any]] = None):
+        self.page_content = page_content
+        self.metadata = metadata or {}
+
+    def __repr__(self):
+        return f"Document(content={self.page_content}, metadata={self.metadata})"
+
+
+class LLMReranker:
```

Comment:
it is almost copy of  reranker_complressor.LLMReranker
why not replace it? or langchain dependend version is going to be used?